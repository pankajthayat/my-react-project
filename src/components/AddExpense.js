import React from "react";
import {NavLink} from "react-router-dom";
import ExpenseForm from "./ExpenseForm"
import {connect} from "react-redux"
import {addExpense} from "../actions/expenses"

export class AddExpense extends React.Component{
    onSubmit=(expense)=>{
        this.props.onSubmit(expense);
        this.props.history.push("/")
    };
    render(){
        return(
            <div>
        <h1>Add ExpenseForm</h1>
        <ExpenseForm 
        onSubmit={this.onSubmit}
        />
        </div>
        )
        
    }
}
// const AddExpense=(props)=>(
//     <div>
//     <h1>Add ExpenseForm</h1>
//     <ExpenseForm 
//     onSubmit={}
//     />
//     </div>
//     );

const mapDispatchToProps=(dispatch)=>({
    onSubmit:(expense)=>dispatch(addExpense(expense))
})

export default connect(undefined,mapDispatchToProps)(AddExpense);