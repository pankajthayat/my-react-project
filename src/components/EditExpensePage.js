import React from "react";
//import {NavLink} from "react-router-dom";
import {connect} from "react-redux";
import ExpenseForm from "./ExpenseForm"
import { removeExpense } from "../actions/expenses"
import {editExpense} from "../actions/expenses"


const EditExpensePage=(props)=>
{
    //console.log(props)
    //console.log("pankaj : "+props.match.params.id);
    return(<div>
      <ExpenseForm
      expense={props.expense}
      onSubmit={(expense)=>{
          // dispach the action to edit the expense and redirect it
          props.dispatch(editExpense(props.expense.id,expense));
          props.history.push("/");
        //  props.history.push("/")
          //console.log("updated ",expense)
      }}
      />
      <button onClick={()=>{
        props.dispatch(removeExpense({id:props.expense.id}))
        }}>Remove</button>
      </div>
  );
  };
  const mapStateToProps=(state,props)=>{
   return {
       expense:state.expenses.find((expense)=>expense.id===props.match.params.id)
   }
  };
  export default connect(mapStateToProps)(EditExpensePage);