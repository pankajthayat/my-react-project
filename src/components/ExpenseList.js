import React from "react"
import {connect} from "react-redux"
import ExpenseListItem from "./ExpenseListItem"
import selectExpenses from "../selectors/expenses";
// const ExpenseList=(props)=>(
//     <div>
//     <h1>ExpenseList</h1>
//      {props.expenses.length}
//     </div>
// )
// () this is where we propvide info about what we want to connect
// we have  a ton  of info in StorageEvent. do we want ot pass all no only subset of that
// so we define a fun here which determine what ifo from store we want our component to access 
// and the store state is passsed as an arg and we return a object...useually these things are from state but we are setying here as key value pair onject for our convienence

//now the connecte compo has access to name prop
// const ConnectedExpenseList=connect((state)=>{
// return{
//     expenses:state.expenses
// }
//})(ExpenseList) ; //connect() means we will get a function in return and we need to call that function. so connect()(); and pass ExpenseList as args

// export default ExpenseList;
//export default ConnectedExpenseList

export const ExpenseList=(props)=>(
    <div>
   {
       props.expenses.length===0?(
        <p>No expense</p>
       ):(
        props.expenses.map((expense)=>(
            <ExpenseListItem key ={expense.id} {...expense}></ExpenseListItem>
        ))
       )
   }
    </div>
)

const mapStateToProps=(state)=>{
    return{
            expenses:selectExpenses(state.expenses,state.filters)
           }
    }
const ConnectedExpenseList=connect(mapStateToProps)(ExpenseList);

export default ConnectedExpenseList;