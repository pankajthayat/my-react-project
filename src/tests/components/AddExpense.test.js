import React from "react";
import { shallow } from "enzyme"
import { AddExpense } from "../../components/AddExpense";

test("should render Add Expense correctly",()=>{
    const onSubmit=jest.fn();
    const history={ push:jest.fn() };
    const wrapper=shallow(<AddExpense onSubmit={ onSubmit } history ={ history } />)
    expect(wrapper).toMatchSnapshot();
})

test("should call onSubmit Correctly ",()=>{
    
})