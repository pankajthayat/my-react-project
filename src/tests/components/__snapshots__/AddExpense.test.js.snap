// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Add Expense correctly 1`] = `
<div>
  <h1>
    Add ExpenseForm
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
